File: Tarea 2/CamiloRivera_primos.c
#include <stdlib.h>
#include <stdio.h>


int main(int argc, char **argv){
	int num=0,d=2,cont=0;
	int primos[2]={0,0};
	
	if(argc!=2){
	printf("Este programa debe recibir un único número!\n EXIT!\n");
	exit(1);
	}
	
	num = atoi(argv[1]);
	
	if(num>1000000 || num<0){
		printf("Entrada no válida. Número debe estar entre 0 y 10^6\n");
		exit(1);
	}
	
	while(d<num && cont<2){
		if(num % d == 0) {
		  primos[cont]=d;
		  num /= d;
		  cont++;
		}
		
		else {
			if(d == 2){
				d = 3;
			}
		  else 
				d += 2;
		}
  }
  
  if (cont>=2 || cont==0){
	  printf("No se ha podido realizar la descomposición prima. \nEl número ingresado no puede expresarse como producto de dos primos\n");
  }else{
	  primos[1]=d;
	  printf("La descomposición prima es:\n %d %d\n",primos[0],primos[1]);
  }	

	return 0;
}

File: Tarea 3/CamiloRivera_primos.py
# encoding: UTF-8
import math
import sys

#Primero se comprueba la validez de la entrada
if len(sys.argv)!=2:
	print "El comando debe ejecutarse con una única entrada numérica"
	quit()

num=int(sys.argv[1])

if (num<0 or num>1000000):
	print "Entrada no válida. Número debe estar entre 0 y 10^6"
	quit()
cont=0
i=2
primos=[0,0,0]
while i<=num:
	if(num%i==0):
		primos[cont]=i
		cont+=1
		num/=i
		if(cont==2):
			break
	else:
		if(i==2):
			i=3
		else:
			i+=2

if(cont!=2 or num!=1):
	print "El número no puede descomponerse en el producto de dos primos. \nLo siento :("
else:
	print `primos[0]`+" "+`primos[1]`
